cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/AbstractNotificationSubscriberService.java

```diff
@@ -163,21 +138,12 @@ public void run() {
       }
     }
 
-    private String loadMessageId() {
-      return Transactionals.execute(transactional, new TxCallable<String>() {
-        @Override
-        public String call(DatasetContext context) throws Exception {
-          return jobQueue.retrieveSubscriberState(topic);
-        }
-      });
-    }
-
     /**
-     * Fetch new notifications and update job queue
+     * Fetches new notifications and configures time for next fetch
      *
      * @return sleep time in milliseconds before next fetch
      */
-    private long processNotifications() {
+    long processNotifications() {
       try {
         final MessageFetcher fetcher = messagingContext.getMessageFetcher();
         String lastFetchedMessageId = Transactionals.execute(transactional, new TxCallable<String>() {
```

Comment:
instead of fetching in a transaction, we can just fetch and store the fetched messages locally without a transaction, then process and persist messageId within a transaction.